<p><img alt="Colaboratory logo" height="45px" src="/img/colab_favicon.ico" align="left" hspace="10px" vspace="0px"></p>



# **Homework 3 (Zixuan Li):**


---


- Please complete each of the problems below
- Show all your code to complete each question
- Provide sentence(s) explaining your solution



It is recommended that you read chapters 11 to prepare for this module!

**Please use the following dataset for problems 1 - 6**





In [1]:
import pandas as pd

URL_Framingham_training='https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Training'
URL_Framingham_test='https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Test'

Framingham_test = pd.read_csv('https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Test')
Framingham_training = pd.read_csv('https://raw.githubusercontent.com/kamranbakhtiari/DESC624/main/Framingham_Training')


##Use training data set to answer questions below

Framingham_test


Index  Sex  Age  Educ  Death
0         4    2   52     2      0
1         6    1   48     1      0
2        10    2   46     3      0
3        14    2   49     2      0
4        16    2   63     1      0
...     ...  ...  ...   ...    ...
2252  11307    2   56     1      0
2253  11313    1   68     1      1
2254  11323    2   50     1      1
2255  11326    2   58     2      0
2256  11328    1   46     3      0

[2257 rows x 5 columns]

**1. Use the training set to run a regression predicting Age, based on Sex and Education.  Obtain a summary of the model.  Do both predictors belong in the model?**





In [2]:
import statsmodels.api as sm

X_train = pd.DataFrame(Framingham_training[['Sex', 'Educ']])
X_train = sm.add_constant(X_train)
y_train = pd.DataFrame(Framingham_training[['Age']])

model01=sm.OLS(y_train, X_train).fit()
model01.summary()

/usr/local/lib/python3.7/dist-packages/statsmodels/tools/_testing.py:19: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.
  import pandas.util.testing as tm


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    Age   R-squared:                       0.017
Model:                            OLS   Adj. R-squared:                  0.017
Method:                 Least Squares   F-statistic:                     67.93
Date:                Sun, 30 May 2021   Prob (F-statistic):           5.58e-30
Time:                        10:56:35   Log-Likelihood:                -29110.
No. Observations:                7953   AIC:                         5.823e+04
Df Residuals:                    7950   BIC:                         5.825e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         56.3600      0.408    138.025      0.000      55.560      57.160
Sex            0.4309      0.213      2.023      0.043       0.013       0.848
Educ          -1.1845      0.103    -11.446      0.000      -1.387      -0.982
==============================================================================
Omnibus:                      297.738   Durbin-Watson:                   1.426
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              148.843
Skew:                           0.136   Prob(JB):                     4.78e-33
Kurtosis:                       2.387   Cond. No.                         12.1
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Both predictors are in the model, the Sex variable has a coefficient of 0.4309 and the Educ variable has a coefficient of -1.1845

**2. Validate the model from the previous exercise**

In [3]:
X_test = pd.DataFrame(Framingham_test[['Sex', 'Educ']])
y_test = pd.DataFrame(Framingham_test[['Age']])
X_test = sm.add_constant(X_test)

model01_test=sm.OLS(y_test, X_test).fit()
model01_test.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:                    Age   R-squared:                       0.020
Model:                            OLS   Adj. R-squared:                  0.019
Method:                 Least Squares   F-statistic:                     22.74
Date:                Sun, 30 May 2021   Prob (F-statistic):           1.67e-10
Time:                        10:56:35   Log-Likelihood:                -8308.5
No. Observations:                2257   AIC:                         1.662e+04
Df Residuals:                    2254   BIC:                         1.664e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const         55.4202      0.784     70.690      0.000      53.883      56.958
Sex            1.0424      0.409      2.548      0.011       0.240       1.845
Educ          -1.1954      0.193     -6.199      0.000      -1.574      -0.817
==============================================================================
Omnibus:                       97.498   Durbin-Watson:                   1.705
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               51.146
Skew:                           0.190   Prob(JB):                     7.83e-12
Kurtosis:                       2.368   Cond. No.                         12.2
==============================================================================

Warnings:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

From the summary tables of linear regression model on both training and test dataset, we can see that the coefficients for constant and education variable are very similar, but the coefficients for the Sex variable are quite different yet still with same direction (both positive).

**3. Use the regression equation to complete this sentence:  "The estimated Age equals..."**


In [4]:
print(f'''The estimated Age equals {round(model01.predict(X_test.head(1))[0], 2)}.''')

The estimated Age equals 54.85.


The code above uses the linear regression model trained on the training dataset to predict the age of the first record in the test dataset, by using Sex and Education as predictors.

**4. Interpret the coefficient for Sex**

In [5]:
print(f'''The coefficient of the Sex variable of the training dataset model is {model01.params['Sex']}.''')

The coefficient of the Sex variable of the training dataset model is 0.4309154906201525.


In [6]:
Framingham_training.Sex.unique()

array([1, 2])

The coefficient of The Sex variable of the training dataset model is 0.43, which means when the Sex variable increase by 1, the dependent variable, which is the Age increase by 0.43. In this case, there're only 2 unique values in the sex variable, which are 1 and 2. Therefore, assuming that 2 represent female and 1 represent male, the coefficient of 0.43 means that the age of a female is 0.43 higher than a male, keeping everything else constant.

**5. Interpret the coefficient for Education.**



In [7]:
print(f'''The coefficient of the Education variable of the training dataset model is {model01.params['Educ']}.''')

The coefficient of the Education variable of the training dataset model is -1.1844730431294532.


In [8]:
Framingham_training.Educ.unique()

array([4, 2, 1, 3])

The coefficient of the Education variable is -1.18, which means when the Education level increase by 1, the age decrease by 1.18, keeping everything else constant.

**6. Find and interpret the value of Standard error.**

In [9]:
model01.bse

const    0.408331
Sex      0.212981
Educ     0.103486
dtype: float64

The standard error of all predictors are shown above, it measures the scatter in the actual data around the estimate regression line.

**7. Find and interpret r2 (adj)**


In [10]:
model01.rsquared_adj

0.016555460752073592

The adjusted r-square value is shown above, it means how much variation can be explained by using this model, which in this case is quite low.

**8. Use your formula to make a prediction of predictor values of your choice.  Apply 95% prediction interval. Interpret the results**

In [11]:
model01.get_prediction(X_test.head(1)).summary_frame(alpha = 0.05)

mean   mean_se  ...  obs_ci_lower  obs_ci_upper
0  54.852894  0.140042  ...      36.40911     73.296678

[1 rows x 6 columns]

The 95% prediction interval is [36.41, 73.3], which means it's 95% confident that the actual age of this record is between 36.41 and 73.3.